datafusion/expr/src/logical_plan/plan.rs

```diff
@@ -1125,6 +1126,13 @@ impl LogicalPlan {
         }
     }
 
+    /// These are invariants to hold true for each logical plan.
+    pub fn assert_invariants(&self) -> Result<()> {
+        assert_unique_field_names(self)?;
+
```

Comment:
I removed the `assert_unions_are_valid` because I was not sure if it should be an invariant, and it caused ~17 test failures.